mapflow/mapflow.py

```diff
@@ -1497,32 +1498,29 @@ def post_processing_callback(self, _: QNetworkReply, processing_name: str) -> No
             url=f'{self.server}/projects/{self.project_id}/processings',
             callback=self.get_processings_callback
         )
-        self.http.get(
-            url=f'{self.server}/projects/{self.project_id}',
-            callback=self.set_processing_limit
-        )
 
-    def set_processing_limit(
-        self,
-        response: Optional[QNetworkReply] = None,
-        response_data: Optional[dict] = None) -> None:
-        """Set the user's processing limit as reported by Mapflow."""
-        if not response and not response_data:
-            raise AssertionError("Either response or response data must be specified")
-        if response:
-            # prefer raw response for no reason
-            response_data = json.loads(response.readAll().data())
-        user = response_data['user']
+    def set_project_name(self, response_data) -> None:
         project_name = response_data['name']
+        if self.plugin_name == 'Mapflow' and self.project_id != 'default':
+            footer = self.tr('Project name: {}').format(project_name)
+            self.dlg.projectNameLabel.setText(footer)
+
+    def set_processing_limit(self) -> None:
+        """Set the user's processing limit as reported by Mapflow."""
+        self.http.get(
+            url=f'{self.server}/user/status',
+            callback=self.set_processing_limit_callback
+        )
 
-        if user['role'] == 'ADMIN':
+    def set_processing_limit_callback(self, response: QNetworkReply) -> None:
+        response_data = json.loads(response.readAll().data())
+        # check here, if user is admin:
+        if response_data.get('admin'):
             self.remaining_limit = 1e5  # 100K sq.km
         else:
-            self.remaining_limit = (user['areaLimit'] - user['processedArea']) * 1e-6
+            self.remaining_limit = response_data.get('remainingLimit')
```

Comment:
default value 0 would be better (to avoid TypeError at least), or maybe somehow show this as error, if we failed to set limit?

Reply:
Also, initialize this field in the __init__, also with 0